src/find_library.cpp

```diff
@@ -69,12 +73,44 @@ std::list<std::string> split(const std::string & value, const char delimiter)
   return list;
 }
 
+// Retrieves a list of paths from the RUNPATH header on Linux.
+// Useful when LD_LIBRARY_PATH is ignored for setcap / setuid executables.
+std::list<std::string> retrieve_runpath_paths()
+{
+#if defined(_WIN32) || defined(__APPLE__)
+  // Return empty list for win / macos
+  return std::list<std::string>();
+#else
+  const ElfW(Dyn) * dyn = _DYNAMIC;
+  const ElfW(Dyn) * runpath = NULL;
+  const char * strtab = NULL;
+
+  for (; dyn->d_tag != DT_NULL; ++dyn) {
+    if (dyn->d_tag == DT_RUNPATH) {
+      runpath = dyn;
+    } else if (dyn->d_tag == DT_STRTAB) {
+      strtab = (const char *)dyn->d_un.d_val;
+    }
+  }
+
+  std::string search_path;
+  if (strtab != NULL && runpath != NULL) {
+    search_path = std::string(strtab + runpath->d_un.d_val);
+  }
+
+  return split(search_path, kPathSeparator);
+#endif
+}
+
 }  // namespace
 
 std::string find_library_path(const std::string & library_name)
 {
-  std::string search_path = get_env_var(kPathVar);
-  std::list<std::string> search_paths = split(search_path, kPathSeparator);
+  std::string search_path_env = get_env_var(kPathVar);
+  std::list<std::string> search_paths = split(search_path_env, kPathSeparator);
+
+  std::list<std::string> search_paths_runpath = retrieve_runpath_paths();
```

Comment:
```suggestion
  const std::list<std::string> search_paths_runpath = retrieve_runpath_paths();
```

Reply:
`splice()` in line 113 manipulates the list, so cannot be const